File: python/pe003.py
'''
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?

Answer = 6857
'''

import time

def prime_factors(n):
	'''
	Returns all the prime factors of a positive integer
	'''
	factors = [2]
	dividend = 3
	while (n > 1):
		if (n%dividend == 0):
			factors.append(dividend)
			n /= dividend
		else:
			dividend += 2

	return factors	


if __name__ == "__main__":
	start = time.time()
	result = prime_factors(600851475143)[-1]
	elapsed = (time.time() - start)
	print "result %s returned in %s seconds." % (result,elapsed)

File: java/pe003.java


public class pe003 {

	public static void main (String[] argv) {
		long factor = 0L;
		long dividend = 2L;
		long n = 600851475143L;
	
		while (n > 1) {
			while (n%dividend == 0) {
				factor = dividend;
				n/= dividend;
			}
			dividend = dividend + 1;
			
			if (dividend*dividend > n) {
				if (n > 1) {
					factor = n;
				}
				break;
			}
		}
		System.out.println(factor);
	}
	
}